sql/rename.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: XisiHuang (cockhuangxh@163.com)
+
+package sql
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/structured"
+)
+
+// RenameDatabase renames the database
+func (p *planner) RenameDatabase(n *parser.RenameDatabase) (planNode, error) {
+	if n.Name == "" {
+		return nil, errEmptyDatabaseName
+	}
+
+	dbDesc, err := p.getDatabaseDesc(string(n.Name))
```

Comment:
let's make this require `root` for now, just like CreateDatabase.
Please add the following comment on the function:
```
privileges: "root" user.
   Notes: postgres requires superuser, db owner, or "CREATEDB".
          mysql >= 5.1.23 does not allow database renames.
```